nucleus/osgi-platforms/osgi-cli-interactive/src/main/java/org/glassfish/osgi/cli/interactive/LocalOSGiShellCommand.java

```diff
@@ -247,43 +266,23 @@ protected int executeCommand()
 
             if (file == null) {
                 System.out.println(strings.get("multimodeIntro"));
-                reader = new ConsoleReader(REMOTE_COMMAND,
+                ConsoleReader reader = new ConsoleReader(REMOTE_COMMAND,
                         new FileInputStream(FileDescriptor.in), System.out,
                         null);
-            } else {
-                printPrompt = false;
-                if (!file.canRead()) {
-                    throw new CommandException("File: " + file
-                            + " can not be read");
-                }
-
-                OutputStream out = new OutputStream() {
-
-                    @Override
-                    public void write(int b) throws IOException {
-                        return;
-                    }
-
-                    @Override
-                    public void write(byte[] b) throws IOException {
-                        return;
-                    }
-
-                    @Override
-                    public void write(byte[] b, int off, int len) throws IOException {
-                        return;
-                    }
-                };
-
-                reader = new ConsoleReader(REMOTE_COMMAND,
-                        new FileInputStream(file), out,
-                        null);
+                reader.setBellEnabled(false);
+                reader.addCompleter(getCommandCompleter());
+                return executeCommands(reader); // NB: wrapper on system.in/out does not need try-with-resource
```

Comment:
sonar should complain about this line now but we then can mark it as false positive as we just wrap `System.in` and `System.out` which should not be closed. Hopefully the comment make recognising this in sonar easier.

Reply:
Actually we will be redoing this a bit, as with 192, this `System.out` no longer reaches the console.

Reply:
Made the comment more general. I assume it still is the case that the general in/out should not be closed by a try-with-resources.